packages/react/src/components/Sticky/util/scroll.ts

```diff
@@ -0,0 +1,25 @@
+export type GetScrollTopInRange = (el: HTMLElement, range: number) => number;
+
+export type ScrollUtils = {
+  getScrollTopInRange: GetScrollTopInRange;
+};
+
+export const getScrollUtils: () => ScrollUtils = () => {
+  const scrollTopElements = new Map<HTMLElement, number>();
```

Comment:
This came up in a conversation elsewhere: there is potentially a memory leak here as `Map`'s keys will be HTML elements and the map will maintain a reference to those keys even if the HTML element is otherwise eligible for garbage collection. Ultimately I think this is okay because the map itself will go out of scope when `getScrollUtils()`'s caller goes out of scope and will release its references then.

Definitely would like more input here though!